### ISSUE TITLE
Unable to get code (auth_ code ) in spotify_token_swap.rb

### ISSUE THREAD

**Issue by sahabe1** [spotify/ios-sdk#929]
Can you please help me how i can get  **auth_code = params[:code]** in spotify_token_swap.rb

**Comment by bih** [comment]
Hi @sahabe1,

You'll get the `code` parameter when Spotify returns the user back to your application. Does this answer your question?

**Comment by sahabe1** [comment]
I can get **code** from callback in application when i login  https://accounts.spotify.com/authorize?nolinks=true&nosignup=true&response_type=code&scope=streaming&utm_source=spotify-sdk&utm_medium=ios-sdk&utm_campaign=ios-sdk&redirect_uri=spotifyiossdkexample%3A%2F%2F&show_dialog=true&client_id=d950416c32d14834bf97ae4a34bf1e37  using this url but in this case i am not able to login in our app getting unsupported url

but when i using  https://accounts.spotify.com/authorize?nolinks=true&nosignup=true&response_type=token&scope=streaming&utm_source=spotify-sdk&utm_medium=ios-sdk&utm_campaign=ios-sdk&redirect_uri=spotifyiossdkexample%3A%2F%2F&show_dialog=true&client_id=d950416c32d14834bf97ae4a34bf1e37  then login is working fine but not receiving code parameter .

in short when i requesting **response_type  = token** the login is working fine but not receiving code value in callback and when requesting **response_type  = code** then login is not working but receiving code value in callback

please help me if i missing anything during app authorization.
